tests/TestCase.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace Tests;
+
+use PHPUnit\Framework\TestCase as BaseTestCase;
+
+abstract class TestCase extends BaseTestCase
```

Comment:
To be honest i followed laravel way to do this + the actual `TestCase` is an abstract too 
- I wanted to see how  laravel put this  
`$this->get('/some-url')'` 
and i found they have like a chain of abstract TestCase classes where they keep lot of helper functions + calling traits everywhere :D  this seems creepy for me, Didn't understand why they are doing this 
+ why testCase is an abstract class anyway ? when i extend it i don't provide any structure to any abstract function so why it's not just a simple class ?

Reply:
@vladimino + The naming i took from laravel too.

Reply:
I thought that frameworks like laravel/symfony are following the best practices so i copy them when possible

Reply:
> I thought that frameworks like laravel/symfony are following the best practices so i copy them when possible

Well, I wouldn’t say so about Laravel. The whole idea about these facades with static calls is corrupt and against all best practices. And I see you took it to this projects too, it's a subject for our future conversations when we're done here ;)

Traits are a powerful way for a so-called horizontal inheritance, and I would suggest using them whenever it is applicable. So far in your application, I couldn't find any suitable functionality, let's see in the future.

But let's talk about this naming explicitly. At first, the `TestCase` from Laravel was introduced to simplify the developer's life and extend the original one from PHPUnit with many helper methods, as you stated above. And you follow this approach with just 2 new methods.

At second, this use case is very specific. I'm questioning even the existence of this abstract class because (at least, for now) I don't see where you can use this `get()` and `post()` methods outside of the router test class. Don't forget applying You Ain't Gonna Need It (**YAGNI**) concept. Don't overcomplicate things until there's a need for this.